lib/EasyPost/Requestor.php

```diff
@@ -68,12 +68,16 @@ private static function _encodeObjects($data)
         } elseif (is_array($data)) {
             $resource = array();
             foreach ($data as $k => $v) {
-                $resource[$k] = self::_encodeObjects($v);
+                if (empty($data)) {
+                    return null;
+                } elseif (!is_null($v)) {
```

Comment:
Compound logic like this is a bit ugly, but I think in this form it would be more like
```suggestion
                if (!is_null($v)) and ($v !== "") and (!is_array($v) or !empty($v)) {
```